tools/oam-converter/pkg/resources/createresources.go

```
@@ -114,5 +118,7 @@ func createChildResources(conversionComponent *types.ConversionComponents, gatew
 		}
 		return virtualServices, destinationRules, authzPolicies, nil
 	}
-	return nil, nil, nil, errors.New("ingress Trait is empty")
+
+	return nil, nil, nil, fmt.Errorf("ingress Trait is empty")
```

Why is it an error if ingress trait is nil? Also, only use fmt.Errorf when you have a format string. errors.New is applicable when you want to return a new error with a string constant.

---

@shahvrushali22